Migration script v2 set remoteWrite remoteTimeout to 5s

pmalek-sumo opened the pull request:
###### Description

Fill in your description here.

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

sumo-drosiek reviewed (commented):
It would be nice to check if value is already set, but LGTM

pmalek-sumo commented:
> It would be nice to check if value is already set, but LGTM

So I've tried a different approach with merging the temporary file that we work on with `remoteWrite` document (with specified `5s` `remoteTimeout`s) but the problem with this is that it might overwrite other values in that tree.

Maybe there's a way of doing this in `yq` but I'm not familiar with it. I'm also not sure if we can iterate over elements with `yq` and then take action in the script.

perk-sumo reviewed (commented):
I think we shouldn't change this value at all.
Just check the user config and in case it is set we should display the INFO that we've changed the default value to 5.

pmalek-sumo commented:
> I think we shouldn't change this value at all.
> Just check the user config and in case it is set we should display the INFO that we've changed the default value to 5.

Changed to just log if `remoteWrite` is set

sumo-drosiek commented:
> Maybe there's a way of doing this in `yq` but I'm not familiar with it. I'm also not sure if we can iterate over elements with `yq` and then take action in the script.

It's possible in `yq`. We did it in upgrade-script for v1

sumo-drosiek commented on deploy/helm/sumologic/upgrade-2.0.0.sh:
```diff
@@ -386,6 +386,18 @@ function add_new_scrape_labels_to_prometheus_service_monitors(){
+    echo
+    info "kube-prometheus-stack.prometheus.prometheusSpec.remoteWrite.[*].remoteTimeout is set"
+    info "Please note that we've set it by default to 5s in 2.0.0"
+  fi
```
We should inform if `remoteTImeout` is not set as well

sumo-drosiek replied:
cc @perk-sumo

sumo-drosiek replied:
or set it

perk-sumo replied:
It is set by default in values.yaml, no need to add this into migrated conf file.

sumo-drosiek commented on tests/upgrade_v2_script/static/prometheus_remote_write_timeout.input.yaml:
```diff
@@ -0,0 +1,117 @@
+          writeRelabelConfigs:
+            - action: keep
+              regex: (?:java_lang_(ClassLoading_(TotalL|Unl|L)oadedClassCount|Compilation_TotalCompilationTime|GarbageCollector_(Collection(Count|Time)|LastGcInfo_(GcThreadCount|duration|(memoryU|u)sage(After|Before)Gc_.*_used))|MemoryPool_(CollectionUsage(ThresholdSupported|_committed|_max|_used)|(Peak|)Usage_(committed|max|used)|UsageThresholdSupported)|Memory_((Non|)HeapMemoryUsage_(committed| ...
+              sourceLabels: [__name__]
```
newline at EOF

perk-sumo reviewed (approved):
👍
